### PR TITLE
[RFC] Test clang-tidy --checks modernize-make-unique

### PR DESCRIPTION
**Pull request by fwyzard**
See the description at https://clang.llvm.org/extra/clang-tidy/checks/modernize-make-unique.html .

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by Dr15Jones** [comment]
looks good to me.

**Comment by alja** [comment]
+1

**Comment by cvuosalo** [comment]
GitHub says, "The table of contents is too big for display." I think this PR has too many files for GitHub to handle well. I don't quite understand what GitHub expects us to do. When the list of files is very long is exactly when the reader needs a table of contents to help find files for review.

**Comment by cvuosalo** [comment]
+1

**Comment by kpedro88** [comment]
@cvuosalo for the big PRs, I'm using `git diff` to inspect them locally (just checking out @fwyzard's branches in a CMSSW area).

**Comment by kpedro88** [comment]
I'm definitely in favor of this check, but it might interact with modernize-use-auto. Should there be a subsequent multi-check PR?

**Comment by makortel** [comment]
> I'm definitely in favor of this check, but it might interact with modernize-use-auto. Should there be a subsequent multi-check PR?

The plan (apparently not very well communicated though, sorry for that) was to show the effect of each option separately now to get feedback, and then deploy the chosen options in smaller PRs around mid-11_2_X (to avoid interfering with backports when changing release cycles).

**Comment by fwyzard** [comment]
I guess for non-controversial checks we could also apply them before forking off 11.2.x ?

**Comment by makortel** [comment]
> I guess for non-controversial checks we could also apply them before forking off 11.2.x ?

I don't mind if we do, but I think it would need some quick planning for which PRs to concentrate the review and when exactly 11_1_0_pre8 gets cut and 11_2_X branched off.

**Comment by santocch** [comment]
+1

**Comment by qliphy** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by jfernan2** [comment]
+1

**Comment by perrotta** [comment]
+reconstruction

**Comment by mommsen** [comment]
+1

**Comment by pohsun** [comment]
+1

File: DQM/Physics/src/SingleTopTChannelLeptonDQM_miniAOD.cc [hunk 1]
```diff
@@ -769,30 +778,29 @@ SingleTopTChannelLeptonDQM_miniAOD::SingleTopTChannelLeptonDQM_miniAOD(const edm
     selectionOrder_.push_back(sel_.at(i).getParameter<std::string>("label"));
     selection_[selectionStep(selectionOrder_.back())] =
         std::make_pair(sel_.at(i),
-                       std::unique_ptr<SingleTopTChannelLepton_miniAOD::MonitorEnsemble>(
-                           new SingleTopTChannelLepton_miniAOD::MonitorEnsemble(
-                               selectionStep(selectionOrder_.back()).c_str(), setup_, consumesCollector())));
+                       std::make_unique<SingleTopTChannelLepton_miniAOD::MonitorEnsemble>(
+
```
Comment (ianna) [inline DQM/Physics/src/SingleTopTChannelLeptonDQM_miniAOD.cc:782]: @fwyzard - this empty line looks odd

File: DQMOffline/RecoB/src/EffPurFromHistos.cc [hunk 2]
```diff
@@ -602,4 +604,6 @@ void EffPurFromHistos::compute(DQMStore::IBooker& ibook) {
   }
 }
 
+#include <memory>
```
Comment (ianna) [inline DQMOffline/RecoB/src/EffPurFromHistos.cc:607]: @fwyzard - it kind of looks random here. Shouldn't it be at the beginning of the file? 
@smuzaffar - it's odd that the `unique_ptr` usage was allowed without its include. Do we run include checker?

File: DQMServices/FwkIO/plugins/DQMRootOutputModule.cc [hunk 3]
```diff
@@ -13,6 +13,8 @@
 // system include files
 #include <algorithm>
 #include <iostream>
+#include <memory>
```
Comment (ianna) [inline DQMServices/FwkIO/plugins/DQMRootOutputModule.cc:16]: @fwyzard - this is a second include, see line 20

File: FastSimulation/SimplifiedGeometryPropagator/plugins/TrackerSimHitProducer.cc [hunk 4]
```diff
@@ -1,3 +1,5 @@
+#include <memory>
```
Comment (ianna) [inline FastSimulation/SimplifiedGeometryPropagator/plugins/TrackerSimHitProducer.cc:1]: @fwyzard - second time

File: GeneratorInterface/AlpgenInterface/plugins/AlpgenSource.cc [hunk 5]
```diff
@@ -1,5 +1,7 @@
 #include <iostream>
 #include <iomanip>
+#include <memory>
```
Comment (ianna) [inline GeneratorInterface/AlpgenInterface/plugins/AlpgenSource.cc:3]: @fwyzard - second time include

File: GeneratorInterface/PartonShowerVeto/src/JetMatching.cc [hunk 6]
```diff
@@ -1,3 +1,5 @@
+#include <memory>
```
Comment (ianna) [inline GeneratorInterface/PartonShowerVeto/src/JetMatching.cc:1]: @fwyzard - I wonder if it relies on an alphabetic include order?

**Review by ianna — COMMENTED** [review]
Looks good in general. I wonder if we should run `llvm-include-order` prior to this one. It looks like it would eliminate the second time includes.

@smuzaffar - FYI

**Comment by mrodozov** [comment]
-note
have it and have a patch for mac release of fireworks

**Comment by alja** [comment]
+1

File: HLTrigger/HLTanalyzers/plugins/HLTInfo.h [hunk 7]
```diff
@@ -109,7 +111,7 @@ HLTInfo::HLTInfo(edm::ParameterSet const& pset, edm::ConsumesCollector&& iC, T&
 
 template <typename T>
 HLTInfo::HLTInfo(edm::ParameterSet const& pset, edm::ConsumesCollector& iC, T& module) : HLTInfo() {
-  hltPrescaleProvider_.reset(new HLTPrescaleProvider(pset, iC, module));
+  hltPrescaleProvider_ = std::make_unique<HLTPrescaleProvider>(pset, iC, module);
```
Comment (Martin-Grunewald) [inline HLTrigger/HLTanalyzers/plugins/HLTInfo.h:114]: What happened to the call to method reset()?

Comment (fwyzard) [reply HLTrigger/HLTanalyzers/plugins/HLTInfo.h:114]: `reset()` releases the old pointed-to object and takes ownership of a new one; assigning a new `unique_ptr` has the same effect

Comment (Martin-Grunewald) [reply HLTrigger/HLTanalyzers/plugins/HLTInfo.h:114]: thanks...

**Comment by Martin-Grunewald** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade
(philosophically)

**Pull request closed without merging**